includes/class-amp-theme-support.php

```diff
@@ -324,10 +324,30 @@ public static function handle_xhr_request() {
 
 		if ( isset( $pagenow ) && 'wp-comments-post.php' === $pagenow ) {
 			// We don't need any data, so just send a success.
-			add_filter( 'comment_post_redirect', function() {
+			add_filter( 'comment_post_redirect', function( $url, $comment ) {
+				// Get theme support.
+				$theme_support = get_theme_support( 'amp' );
+
+				// Add the comment ID to the URL to force AMP to refresh the page.
+				$url = add_query_arg( 'comment', $comment->comment_ID, $url );
+
+				// Send redirect header if amp-live-list has been opted-out.
+				if ( empty( $theme_support['comments_live_list'] ) ) {
+					header( 'AMP-Redirect-To: ' . $url, true );
+				}
+				// Create a success message to display to the user.
+				if ( '1' === (string) $comment->comment_approved ) {
+					$message = __( 'Your comment has been posted and has been approved.', 'amp' );
+				} else {
+					$message = __( 'Your comment has been posted, and is awaiting moderation.', 'default' );
+				}
+
+				$message = apply_filters( 'amp_comment_submitted_message', $message, $comment );
+
 				// We don't need any data, so just send a success.
-				wp_send_json_success();
-			}, PHP_INT_MAX );
+				wp_send_json_success( $message );
```

Comment:
We should probably apply kses to this to only allow the allowed amp-mustache markup.